app/models/cart.rb

```diff
@@ -0,0 +1,33 @@
+class Cart < ApplicationRecord
+  belongs_to :user
+  has_many :product_carts, dependent: :destroy
+  has_many :products, through: :product_carts, foreign_key: :cart_id
+
+  def add_product!(product_id)
+    if product_ids.include?(product_id)
+      product_carts.find_by(product_id: product_id).increment!(:quantity)
```

Comment:
you can make use of ActiveRecord's counter cache to make this nicer :-)
https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html

Reply:
Sounds good, thanks for the suggestion. I'll check it out!